## Problem 20: Factorial Digit Sum

$n!$ means $n × (n − 1) × ... × 3 × 2 × 1$

For example, $10! = 10 × 9 × ... × 3 × 2 × 1 = 3628800$,
and the sum of the digits in the number $10!$ is $3 + 6 + 2 + 8 + 8 + 0 + 0 = 27$.

Find the sum of the digits in the number $100!$

In [1]:
# Python can handle large numbers by default. Can use the factorial function from the math package

import math

factorial_number = 100

factorial = math.factorial(factorial_number)

solution = sum(int(s) for s in str(factorial))

print(f'There are {len(str(factorial)):,} digits in 100! and their sum is {solution:,}.')

There are 158 digits in 100! and their sum is 648.


### Problem 20 Solution:

There are 158 digits in 100! and their sum is 648.